torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -3036,6 +3038,39 @@ void TensorExprKernel::bindConstant(const torch::jit::Value* v) {
   bufs_[v] = buf;
 }
 
+void* allocBuf(size_t size) {
+  void* bp = (void*)malloc(size);
+  return bp;
+}
+
+void TensorExprKernel::preAllocIntermediateBufs(
+    const std::unordered_set<BufPtr> intem_bufs) {
+  void* bp;
+  for (auto buf : intem_bufs) {
```

Comment:
Iterating over unordered_set yields non-deterministic results - i.e. the order in which the buffers get into constants_ is different each time we run. While this is correct each time, it's often quite annoying to deal with that when, e.g., we need to debug something, because execution might be different from run to run. It is usually recommended to put all elements into a vector and sort it before iterating over them to avoid this.